app/services/proxy_status/logs.rb

```diff
@@ -0,0 +1,18 @@
+module ProxyStatus
+  class Logs < Base
+    def api
+      ProxyAPI::Logs.new(:url => proxy.url)
+    end
+
+    def logs
+      fetch_proxy_data do
+        ::SmartProxies::LogBuffer.new(api.all((Setting[:proxy_expired_logs].to_i rescue 0) || 0))
```

Comment:
Again, this shouldn't be a Setting, I think it would be best to have this as an instance variable for the logs status object